tsconfig.json

```diff
@@ -2,24 +2,31 @@
 	"compilerOptions": {
 		"allowJs": true,
 		"checkJs": true,
-		"noEmit": true,
-		"module": "es2015",
+		"emitDeclarationOnly": true,
+		"declaration": true,
+		"module": "esnext",
 		"moduleResolution": "node",
 		"target": "es5",
-		"baseUrl": "packages",
-		"paths": {
-			"@tannin/*": [ "*/index.js" ],
-			"tannin": [ "tannin/index.js" ]
-		}
+		"composite": true,
+
+		"strict": true,
+		"strictNullChecks": true,
+		"noImplicitAny": true,
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noImplicitReturns": true,
+		"noFallthroughCasesInSwitch": true,
+
+		/* This needs to be false so our types are possible to consume without setting this */
+		"esModuleInterop": false,
+		"isolatedModules": true
 	},
-	"include": [
-		"packages"
-	],
+	"files": [],
 	"exclude": [
 		"**/build",
 		"**/dist",
 		"**/__tests__",
 		"**/benchmark",
-		"node_modules"
+		"**/node_modules"
```

Comment:
Not sure if it might be related to the errors I mentioned earlier, but considering if this configuration exists as a base, and is effectively evaluated as in the context of individual packages, then the exclude patterns would probably need to be considered as relative to the package's root. In other words, we're no longer considering excludes from the _root of the repository_, but from the root of the package. Therefore, the `**/` in each of these should not be necessary (at worst, causing those folders not to be ignored)?

Reply:
> if this configuration exists as a base, and is effectively evaluated as in the context of individual packages, then the exclude patterns would probably need to be considered as relative to the package's root

Paths are relative to wherever they exist in configuration, which is nice for defining these root excludes and is also the reason `rootDir` is declared for each project.

> All relative paths found in the configuration file will be resolved relative to the configuration file they originated in.

https://www.typescriptlang.org/en/tsconfig#extends